service/history/outbound_queue_standby_task_executor.go

```diff
@@ -115,7 +117,15 @@ func (e *outboundQueueStandbyTaskExecutor) Execute(
 		}
 		// If there was no error from Access nor from the accessor function, then the task
 		// is still valid for processing based on the current state of the machine.
-		return nil, consts.ErrTaskRetry
+		// The post action function needs a non-nil object so the task is retried properly.
+		return struct{}{}, nil
+	}
+
+	// MaxInt64 duration is 290+ years, so this is effectively equivalent to never
+	// discarding the task.
+	discardDelay := time.Duration(math.MaxInt64)
+	if e.config.OutboundStandbyDiscardTaskMissingEvents(task.GetType()) {
+		discardDelay = e.config.StandbyTaskMissingEventsDiscardDelay(task.GetType())
```

Comment:
Can we just have a single discard delay config for this that's per source namespace and destination?